### ISSUE TITLE
Add test for statusForException()

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2083]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#2083|
|Type|enhancement|
|Created|2006-09-12 23:51:08Z|

web2.dav.http.statusForException() needs a sanity test

Attachments:

* [test_http.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8c4/8c4178948b7b02a62a91106c986bd5a70847c4c7/cd809d65406f4f145a39e5b7d1e2a36f35308c8f.py) (3530 bytes) - added by wsanchez on 2006-09-27 16:44:57Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2083 2083
type__enhancement enhancement
reporter__wsanchez wsanchez
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2_dav web2.dav
keywords__ 
time__1158105068000000 1158105068000000
changetime__1159462709000000 1159462709000000
version__None None
owner__ 
cc__wsanchez
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez set owner to @glyph|
|-|-|

See attached file for a set of tests.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @dreid|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_4"></a>|@dreid set owner to @wsanchez|
|-|-|

You might as well add a test for unknown exception as well.  

Also I'm somewhat confused about statusForFailure's usage.  Obviously it tries to figure out the HTTP status code for a given failure.  But it's handling of HTTPError seems strange.  Is statusForException ever used to construct a new HTTPError? If so information from HTTPErrors could be lost if it were used improperly.  Or is statusForFailure _only_ to be used when constructing the XML for a MultiStatusResponse?

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez set status to `assigned`|
|-|-|

statusForFailure() returns an HTTP status code which corresponds to a Failure.  If the Failure contains an HTTPError, the status code of that HTTPError is the appropriate result.  Yes, if you have an HTTPError, and you grap the status code from that and toss the rest, you will lose information.  I don't find that surprising.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_7"></a>|@wsanchez set owner to @dreid<br>@wsanchez set status to `new`|
|-|-|

New test checks for unknown exception and verifies that it re-raises.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_8"></a>|@dreid set owner to @wsanchez|
|-|-|

Ok reviewed

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_9"></a>|@wsanchez set status to `closed`|
|-|-|

(In [18310]) Add unit tests for statusForFailure().
Reviewed by dreid.
Fixes [#2083](https://github.com/twisted/twisted/issues/2083).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
